src/Polly.Core/Utils/Pipeline/DelegatingComponent.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+
+namespace Polly.Utils.Pipeline;
+
+/// <summary>
+/// A component that delegates the execution to the next component in the chain.
+/// </summary>
+internal sealed class DelegatingComponent : PipelineComponent
+{
+    private readonly PipelineComponent _component;
+
+    public DelegatingComponent(PipelineComponent component) => _component = component;
```

Comment:
```suggestion
    public DelegatingComponent(PipelineComponent component, bool aotSupported) => _component = component;
```

And pass it from outside? This would allow testing both paths.

Reply:
I don't think it's worth it to remove the need to add `[ExcludeFromCodeCoverage]` for one branch of one line of code and then need to try and suppress the IL3054 warning.

Reply:
Ok, haven't realized this. Let's leave it as it is.